PLC Development/DrivePLC/POUs/MAIN.TcPOU

```diff
@@ -1,59 +1,389 @@
+      </Implementation>
+    </Method>
+    <Action Name="PROG" Id="{a83d747a-28b8-44b3-812d-dbc6c7a4a77b}">
+      <Implementation>
+        <ST><![CDATA[CASE iState OF
+	0:	//CHECK AXIS ID's are correct. Gearing limits currently requires the assigned array index of an axis to match the NC ID.
+		//May be possible to negate this by changing axes to GVLs so that they can be seen within the drives function - CASE 0 NO LONGER REQ
+		iState:=1;
+		FOR i:=1 TO GVL.axisNumMax BY 1 DO
+			IF GVL.axes[i].Axis.NcToPlc.AxisId <> i THEN
+				iState:=999;
+			END_IF
+		END_FOR
+		
+	1:	//Gear out all axes to begin
+		FOR i:=1 TO 8 BY 1 DO
+			GearOut(i:=i);
+		END_FOR
+		iState:= 2;
+	2:	//Wait for gear out
+		IF CheckDone(iLow:=1, iHigh:=8) THEN
+			iState:=3;
+		END_IF
+	3:	//Power up axes (and set limits true as no hardware connected)
+		FOR i:=1 TO 8 BY 1 DO
+			PowerUp(i:=i);
+		END_FOR
+		iState:= 10;
+		
+	10:	//Set position of virtual gap axes based on position. In reality we would do some homing sequence of blades first then set the position
+		SlitPositions(North:=1,South:=2,East:=3,West:=4,VG:=5,VC:=6,HG:=7,HC:=8);
+		iState:=11;
+		
+	11:	//Wait for position set
+		IF CheckDone(iLow:=5,iHigh:=8) THEN
+			iState:=12;
+		END_IF		
+		
+	12:	//GEAR IN SLIT GAP AND CENTRE
+		SlitSetup(North:=1,South:=2,East:=3,West:=4,VG:=5,VC:=6,HG:=7,HC:=8);
+		iState:=13;
+	
+	13:	//WAIT FOR DONE
+		IF CheckDone(iLow:=1, iHigh:=4) THEN
+			iState:=14;
+		END_IF	
+
+	14: //ALL SETUP!	
+			IF bDisableMC THEN
+				iState:=50;
+			END_IF
+	
+	50:
+		FOR i:=1 TO 8 BY 1 DO
+			GearOut(i:=i);
+		END_FOR
+		iState:=51;
+	51:
+		IF CheckDone(iLow:=1, iHigh:=8) THEN
+			iState:=100;
+		END_IF					
+			
+	100:	//DO NOTHING
+
+
+		
+		IF NOT bDisableMC THEN
+			iState:=1;
+		END_IF
+
+		
+	
+	999: //ERROR
+		//INOP STATE	
+END_CASE]]></ST>
+      </Implementation>
+    </Action>
+    <Action Name="SIMULATE" Id="{3ab21edf-3a61-475f-a4eb-e4bcda8bf9cd}">
+      <Implementation>
+        <ST><![CDATA[(*
+Simulate axes if bool set TRUE
+*)
+FOR i:=1 TO GVL.axisNumMax BY 1 DO
+	IF bSimulate[i] THEN
+		simulateStage(position:=GVL.axes[i].Axis.NcToPlc.ActPos, iAxis:=i, simulated:=simulated[i]);
+		GVL.axes[i].MCSignalRef.Level:=simulated[i].absSw;
+	END_IF
+END_FOR
+]]></ST>
+      </Implementation>
+    </Action>
+    <Method Name="simulateStage" Id="{deb827a7-c982-4c31-9561-f2bd7c832d23}" FolderPath="Simulate Methods\">
+      <Declaration><![CDATA[(*
+	Basic virtual axis simulation.
+	Will trigger limit switches beyond set points
+	Will trigger absolute switch between given set points
+*)
+METHOD simulateStage : BOOL
+VAR_INPUT
+	position	:	LREAL;		//Axis current position
+	iAxis		:	UINT;		//Axis index
+END_VAR
+VAR_IN_OUT
+	simulated	:	ST_SimulateStage;	//Struct to hold simulation parameters
+END_VAR
+]]></Declaration>
+      <Implementation>
+        <ST><![CDATA[IF position <simulated.fHighLim THEN
+	Limits.FwLimit[iAxis]:=TRUE;
```

Comment:
You can do some contraction here e.g. `Limits.FwLimit[iAxis] := position < simulated.fHighLim` to replace the whole if statement. Is it more readable? Discuss...